kvbc/src/kvbc_app_filter/kvbc_app_filter.cpp

```diff
@@ -387,48 +645,77 @@ string KvbAppFilter::readBlockRangeHash(BlockId block_id_start, BlockId block_id
   return computeSHA256Hash(concatenated_update_hashes);
 }
 
-string KvbAppFilter::readEventGroupRangeHash(EventGroupId event_group_id_start, EventGroupId event_group_id_end) {
-  auto opt = rostorage_->getLatest(kvbc::categorization::kExecutionEventGroupIdsCategory, client_id_);
-  if (not opt) {
+string KvbAppFilter::readEventGroupRangeHash(EventGroupId event_group_id_start) {
+  uint64_t public_start = getValueFromLatestTable(kPublicEgIdKeyOldest);
+  uint64_t public_end = getValueFromLatestTable(kPublicEgIdKeyNewest);
+  uint64_t private_start = getValueFromLatestTable(client_id_ + "_oldest");
+  uint64_t private_end = getValueFromLatestTable(client_id_ + "_newest");
+  if (!public_start && !private_start) {
     std::stringstream msg;
-    msg << "An event group for trid \"" << client_id_ << "\" cannot be found";
+    msg << "Event groups do not exist for client: " << client_id_ << " yet.";
+    LOG_ERROR(logger_, msg.str());
     throw std::runtime_error(msg.str());
+  } else if (!public_start) {
+    ConcordAssertEQ(public_start, public_end);
```

Comment:
Nononono :) From an earlier comment, when we prune, this is not true.

Reply:
My bad! 🙈 
Updated.